infra/lint.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/sh
+
+set -e
+
+POETRY_RUN="poetry run"
+
+bandit () {
+  $POETRY_RUN bandit -lll --recursive src
+}
+black () {
+  $POETRY_RUN black ${check:+--check} src tests
+}
+detect_secrets () {
+  # Scan only files fixed into the repo, omit poetry.lock
+  SECRETS_TO_SCAN=`git ls-tree --full-tree -r --name-only HEAD | grep -v poetry.lock`
+  $POETRY_RUN detect-secrets-hook $SECRETS_TO_SCAN --baseline .secrets.baseline
+}
+isort () {
+  $POETRY_RUN isort ${check:+--check-only} src
+}
+pylint () {
+  $POETRY_RUN pylint src tests
+}
+mypy () {
+  $POETRY_RUN mypy src
+}
+yamllint () {
+  $POETRY_RUN yamllint -c .yamllint config/*
+}
+all () {
+  echo "running black"
+  black
+  echo "running isort"
+  isort
+  echo "running mypy"
+  mypy
+  echo "running pylint"
+  pylint
+  echo "running yamllint"
+  yamllint
+  echo "running bandit"
+  bandit
+  echo "running detect_secrets"
+  detect_secrets
+}
+
+usage () {
+  echo "Usage: infra/lint.sh [OPTION]"
+  echo " run linting checks"
+  echo "Options":
+  echo "  bandit"
+  echo "  black [--fix]"
+  echo "  detect-secrets"
+  echo "  isort [--fix]"
+  echo "  mypy"
+  echo "  pylint"
+  echo "  yamllint"
+}
+
+subcommand='';
+check="true"
+if [ -z $1 ]; then
+  all
+else
+  subcommand=$1; shift
+  case $subcommand in
+    "black" | "isort")
+      case $1 in
+        "--fix")
+          check=""
+        ;;
+      esac
+      case $subcommand in
+        "isort") isort;;
+        "black") black;;
```

Comment:
You don't pass the `$check` argument here ?

Reply:
I think the `$check` param argument will be picked up when it's set above. I guess it acts like a global?

Tbh, I'm not the best Bash author, so I'd welcome a PR to make this bash script nicer. I was also thinking of rewriting this in Python -- we don't _need_ to torture ourselves with Bash.